hudi-flink-datasource/hudi-flink/src/test/java/org/apache/hudi/source/stats/TestExpressionEvaluator.java

```diff
@@ -103,8 +108,9 @@ void testEqualTo() {
     equalTo.bindColStats(indexRow6, queryFields(2), rExpr);
     assertFalse(equalTo.eval(), "12 <> null");
 
-    equalTo.bindVal(new ValueLiteralExpression(null, DataTypes.INT()));
-    assertFalse(equalTo.eval(), "null <> null");
+    assertThrows(
+        IllegalStateException.class,
```

Comment:
Not quite understand why it throws here?

Reply:
There is a preprocessing in `bindCall` which would translate to `AlwaysFalse` evaluator instead of EqualsTo/NotEqualsTo/In...
So the other evaluator could not support null val.

Reply:
There is no regression for the other evaluators to support nulls right? Can we keep that part, I think the AlwaysFalse is an improvement but each evaludator should also handle the nulls correctly.

Reply:
Make sense. 
That's what I've done in the first commit, which is removed in the second commit.
I revert this part now.